### PR TITLE
add fish shell completion

### LINKED ISSUE #4208: Add fish shell completions
## Feature Request
Add fish shell completions. Currently only bash/zsh are supported.

### What problem are you trying to solve?
Adding completions to the very popular fish shell

### How should the problem be solved?

Add support for `linkerd completion fish`.

### PR DESCRIPTION
**Pull request by WLun001**
fixes #4208

### CODE REVIEW

**Comment by WLun001** [comment]
Not sure how to fix the failed CI checks. Could be change of dependencies? Please advise

**Comment by adleong** [comment]
@WLun001 that's right.  since you updated the go dependencies, you'll need to run the `bin/update-go-deps-shas` script to update the Dockerfiles to use the new go dependency images.

File: cli/cmd/completion.go [hunk 1]
```diff
@@ -29,15 +29,22 @@ func newCmdCompletion() *cobra.Command {
   source <(linkerd completion zsh)
 
   # zsh on osx / oh-my-zsh
-  linkerd completion zsh > "${fpath[1]}/_linkerd"`
+  linkerd completion zsh > "${fpath[1]}/_linkerd"
+
+  # Fish:
```
Comment (kleimkuhler) [inline cli/cmd/completion.go:34]: Small suggestion, but just keeping it consistent with the above help text

```suggestion
  # fish
```

File: cli/cmd/completion.go [hunk 2]
```diff
@@ -29,15 +29,22 @@ func newCmdCompletion() *cobra.Command {
...
+  linkerd completion fish | source
+
+  # fish  To load completions for each session, execute once:
```
Comment (kleimkuhler) [inline cli/cmd/completion.go:37]:
```suggestion
  # To load fish shell completions for each session, execute once:
```

**Pull request merged**